pages/api/articles/[id].ts

```diff
@@ -0,0 +1,14 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import Article from "../../../models/Article";
+import articles from "./articles.mock";
+
+export default function handler(req: NextApiRequest, res: NextApiResponse) {
+  const { id } = req.query;
+  let article: Article | undefined;
+
+  article = articles.find((a) => a.id == id);
```

Comment:
No need to use `let` here, `const article: Article | undefined = articles.find((a) => a.id == id);` will also have the same effect.